File: go-client-sdk/pkg/models/operations/cancelrun.go
package operations

import (
	"github.com/speakeasy-sdks/nolan-sdks/pkg/models/shared"
)

type CancelRunPathParams struct {
	ProjectID string `pathParam:"style=simple,explode=false,name=projectId"`
	RunID     string `pathParam:"style=simple,explode=false,name=runId"`
}

type CancelRunRequest struct {
	PathParams CancelRunPathParams
}

type CancelRunResponse struct {
	ContentType              string
	StatusCode               int
	TsoaErrorResponsePayload *shared.TsoaErrorResponsePayload
}

File: typescript-client-sdk/src/sdk/models/operations/cancelrun.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";


export class CancelRunPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=projectId" })
  projectId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=runId" })
  runId: string;
}

export class CancelRunRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: CancelRunPathParams;
}

export class CancelRunResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  tsoaErrorResponsePayload?: shared.TsoaErrorResponsePayload;
}